docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -59,46 +62,254 @@ The following properties are supported in ``qlpack.yml`` files.
+
+For example, if a ``qlpack.yml`` file contains the following dependencies:
+
+.. code-block:: yaml
+
+   dependencies:
+     codeql/cpp-all: ^0.1.2
+     my-user/my-lib: ^0.2.3
+     other-dependency/from-source: "*"
+
+The ``codeql-pack.lock.yml`` file will contain something like the following:
+
+.. code-block:: yaml
+
+   dependencies:
+     codeql/cpp-all:
+       version: 0.1.4
+     my-user/my-lib:
+       version: 0.2.4
+     my-user/transitive-dependency:
+       version: 1.2.4
+
+The ``codeql/cpp-all`` dependency is locked to version 0.1.4. The ``my-user/my-lib`` dependency is locked to version 0.2.4. The ``my-user/transitive-dependency``, which is a transitive dependency and is not specified in the ``qlpack.yml`` file, is locked to version 1.2.4. The ``other-dependency/from-source`` is absent from the lock file since it is resolved from source. This dependency must be available in the same CodeQL workspace as the pack.
+
+.. _custom-codeql-packs:
+
+Examples of custom CodeQL packs
+-------------------------------
+
+When you write custom queries or tests, you should save them in custom CodeQL packs. For simplicity, try to organize each pack logically. For more information, see `CodeQL pack structure <#codeql-pack-structure>`__. Save files for queries and tests in separate packs and, where possible, organize custom packs into specific folders for each target language. This is particuarly useful if you intend to publish your CodeQL packs so they can be shared with others or used in GitHub `Code scanning <https://docs.github.com/en/code-security/secure-coding/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning>`__.
+
+CodeQL packs for custom queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A custom CodeQL pack for queries must include a ``qlpack.yml`` file at the pack root, containing ``name``, ``version``, and ``dependencies`` properties. If the pack contains query suites, you can use the ``suites`` property to define their location. Query suites defined here are called "well-known" suites, and can be used on the command line by referring to their name only, rather than their full path. For more information about query suites, see ":doc:`Creating CodeQL query suites <creating-codeql-query-suites>`."
```

Comment:
The first sentence here seems a bit redundant with the "qlpack.yml properties" section (and custom CodeQL packs presumably must also contain `codeql-pack.lock.yml` files?)